app/controllers/posts_controller.rb

```diff
@@ -1,28 +1,30 @@
 class PostsController < ApplicationController
   def index
     @user = User.all.find(params[:user_id])
-    @last_user = User.last
     @posts = []
+    @comments = Comment.all
+    @likes = Like.all
     Post.all.each do |post|
-      @posts.push(post.title) if post.author_id.to_s == params[:user_id]
+      @posts.push(post) if post.author_id.to_s == params[:user_id]
     end
   end
 
   def show
     @user = User.all.find(params[:user_id])
+    @users = User.all
     @post = Post.find_by(author: params[:user_id], id: params[:id])
     @recent_comments = Post.limit(3).where(author_id: params[:id]).order(created_at: :desc)
+    @comments = Comment.all
+    @likes = Like.all
```

Comment:
- [ ] We don't need the user in the post show page, because we are not displaying any users. Kindly remove it.
- [ ] You could refactor `@post = Post.find_by(author: params[:user_id], id: params[:id])` to `@post Post.find(params[:post_id])`
- [ ] Now that you have the post, to get recent comments you could do:
```ruby
@recent_comments = @post.comments.limit(3).order(created: :desc)
@comments = @post.comments
@likes = @post.likes
```